src/glyph.rs

```diff
@@ -0,0 +1,166 @@
+use glib::translate::*;
+use pango_sys;
+use GlyphItem;
+use GlyphString;
+use Item;
+
+impl GlyphString {
+    pub fn num_glyphs(&self) -> i32 {
+        unsafe { (*self.to_glib_none().0).num_glyphs }
+    }
+
+    pub fn glyph_info(&self) -> Vec<GlyphInfo> {
+        if self.num_glyphs() < 0 {
+            return Vec::new();
+        }
+        let num_glyphs = self.num_glyphs() as usize;
+        unsafe {
+            let glyphs: *mut pango_sys::PangoGlyphInfo = (*self.to_glib_none().0).glyphs;
+            FromGlibContainer::from_glib_none_num(glyphs, num_glyphs)
+        }
+    }
+}
+
+impl GlyphItem {
+    pub fn item(&self) -> Item {
+        unsafe { from_glib_none((*self.to_glib_none().0).item) }
+    }
+
+    pub fn glyph_string(&self) -> GlyphString {
+        unsafe { from_glib_none((*self.to_glib_none().0).glyphs) }
+    }
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+#[repr(C)]
+pub struct GlyphInfo(*mut pango_sys::PangoGlyphInfo);
```

Comment:
This doesn't have to be repr(C) or does it?

Reply:
Nope, removed